lib/bojangles.rb

```diff
@@ -83,18 +83,18 @@ def cache_error_messages!(current_errors)
   def cached_error_messages
     if File.file? 'error_messages.json'
       JSON.parse File.read('error_messages.json')
-    else []
+    else {}
     end
   end
 
   # Only the first two lines of an error message need to match for us to decide
   # that we've already sent it.
   def compare_errors(current_errors, old_errors)
     new_errors = current_errors.reject do |error|
-      old_errors.any? { |old| error.lines.first(2) == old.lines.first(2) }
+      old_errors.keys.any? { |old| error[:message].lines.first(2) == old.lines.first(2) }
     end
     resolved_errors = old_errors.reject do |error|
-      current_errors.any? { |new| error.lines.first(2) == new.lines.first(2) }
+      current_errors.any? { |new| error.lines.first(2) == new[:message].lines.first(2) }
```

Comment:
GitHub's _syntax-highlighting_ isn't a fan of using `new` as a variable name, but this fine, I'm pretty sure.

Reply:
My understanding is that it's only interpreted as a keyword when followed by a class name.